reader/reader/core/reader_service.py

```diff
@@ -91,7 +96,14 @@ def get_many(self, request: GetManyRequest) -> Dict[str, Model]:
                 result = self.database.get_many(
                     fqids, mapped_fields_per_fqid, request.get_deleted_models,
                 )
-        return result
+
+        # change mapping fqid->model to collection->id->model
+        final: Dict[str, Dict[str, Model]] = {}
+        for fqid, model in result.items():
+            collection = final.setdefault(collection_from_fqid(fqid), {})
```

Comment:
Please have a look at `from collections import defaultdict` ;)

Reply:
Debatable if that's so much nicer, but I've changed it for now :D at least now the two lines can be one with a good conscience